ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java

```diff
@@ -288,32 +343,46 @@ public static void attestations(BeaconState state, BeaconBlock block) {
             attestation.getData().getJustified_epoch().equals(state.getPrevious_justified_epoch()));
       }
 
-      try {
-        checkArgument(
-            attestation.getData().getJustified_block_root()
-                == get_block_root(
-                    state, get_epoch_start_slot(attestation.getData().getJustified_epoch())));
-      } catch (Exception e) {
-        throw new AssertionError();
-      }
-
+      // - Verify that attestation.data.justified_block_root is equal to
+      //     get_block_root(state, get_epoch_start_slot(attestation.data.justified_epoch)).
       checkArgument(
-          attestation.getData().getLatest_crosslink_root()
-                  == state
-                      .getLatest_crosslinks()
-                      .get(attestation.getData().getShard().intValue())
-                      .getShard_block_root()
-              || attestation.getData().getShard_block_root()
-                  == state
-                      .getLatest_crosslinks()
-                      .get(attestation.getData().getShard().intValue())
-                      .getShard_block_root());
-
+          Objects.equals(
+              attestation.getData().getJustified_block_root(),
+              get_block_root(
+                  state, get_epoch_start_slot(attestation.getData().getJustified_epoch()))));
+
+      // - Verify that either attestation.data.latest_crosslink_root or
+      // attestation.data.shard_block_root
+      //     equals state.latest_crosslinks[shard].shard_block_root.
+      checkArgument(
+          attestation
+                  .getData()
+                  .getLatest_crosslink_root()
+                  .equals(
+                      state
+                          .getLatest_crosslinks()
+                          .get(attestation.getData().getShard().intValue())
+                          .getShard_block_root())
+              || attestation
+                  .getData()
+                  .getShard_block_root()
+                  .equals(
+                      state
+                          .getLatest_crosslinks()
+                          .get(attestation.getData().getShard().intValue())
+                          .getShard_block_root()));
+
+      // - Verify bitfields and aggregate signature
       checkArgument(verify_bitfields_and_aggregate_signature(attestation, state));
 
-      checkArgument(
-          attestation.getData().getShard_block_root() == ZERO_HASH); // [TO BE REMOVED IN PHASE 1]
+      // - Verify that attestation.data.shard_block_root == ZERO_HASH
+      // TO BE REMOVED IN PHASE 1
+      checkArgument(attestation.getData().getShard_block_root() == ZERO_HASH);
```

Comment:
should this be: `checkArgument(attestation.getData().getShard_block_root().equals(ZERO_HASH));`?

Reply:
Yep, thanks for noticing. Thought I fixed all those. Fixing this and one other occurrence of similar incorrect behavior in this file now.